### PR TITLE
Add fuzz tests for JWS, JWE & JWT parsing

### PR DESCRIPTION
**Pull request by mcpherrinm**
This adds fuzz testing for the different parsing functions in go-jose.

The seed corpus were generated by Claude/Fable 5/Opus4.8. It was prompted to run the tests with coverage, then introspect the code to add more coverage. It ran several iterations increasing the coverage. I hope that hybrid of LLM and fuzzing helps ensure high coverage of edge cases.

I'm not adding any explicit fuzzing in CI, but they'll run as normal Go tests.

### CODE REVIEW

File: fuzz/jws_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		}
+		sig, err = jose.ParseSignedJSON(data, allSignatureAlgorithms)
+		if err == nil {
+			sig.FullSerialize()
+			_, _ = sig.CompactSerialize()
+		}
+	})
+}
+
+// FuzzJWSParseDetached has a different signature so gets its own test from FuzzJWSParse, but doesn't have much extra
```
Comment (jsha) [inline fuzz/jws_test.go]:
```suggestion
// FuzzJWSParseDetached is like FuzzJWSParse but calls ParseDetached, which takes two separate args.
```

**Review by jsha — APPROVED** [review]
Looks good with one comment diff.

File: fuzz/jws_test.go (hunk 1 above)
Comment (mcpherrinm) [reply fuzz/jws_test.go]: Ooop, I'd meant to finish that sentence. Thanks!
